Bottom: f0f6e5f7b485a533c3502ef0c29f565fe23d4762
Top:    03ee1ce837c1b6da9a8092ceb1c0b5e93452e8da
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 00:31:29 +0100

Try optimising polygon operations by keeping an rtree with their contours

InsertHoles() already creates an rtree of contours to do its internal
processing, and it is possible that intersect() could be sped up by
its existance as well.




---

diff --git a/src/polyarea.h b/src/polyarea.h
index 42acc1a..819b074 100644
--- a/src/polyarea.h
+++ b/src/polyarea.h
@@ -128,6 +128,7 @@ struct POLYAREA
 {
     POLYAREA *f, *b;
     PLINE *contours;
+    void *contour_tree;
 };
 
 BOOLp poly_M_Copy0(POLYAREA ** dst, const POLYAREA * srcfst);
